Gravity/Gravity/DAL/RSAPI/RsapiDao.Update.cs

```diff
@@ -12,211 +12,149 @@ namespace Gravity.DAL.RSAPI
 {
 	public partial class RsapiDao
 	{
-		#region UPDATE Protected Stuff
-		protected void UpdateRdos(params RDO[] rdos)
+		private void UpdateSingleObjectFields<T>(IList<T> objectsToUpdate, bool recursive) where T : BaseDto
 		{
-			rsapiProvider.Update(rdos).GetResultData();
-		}
+			var singleObjectProperties =
+				typeof(T).GetProperties()
+				.Where(x => x.GetCustomAttribute<RelativityObjectFieldAttribute>()?.FieldType == RdoFieldType.SingleObject);
 
-		protected void UpdateRdo(RDO theRdo)
-		{
-			rsapiProvider.UpdateSingle(theRdo);
-		}
+			foreach (var propertyInfo in singleObjectProperties)
+			{
+				var childObjectsToUpdate = objectsToUpdate
+					.Select(objectToUpdate => (BaseDto)objectToUpdate.GetPropertyValue(propertyInfo.Name))
+					.Where(x => x != null);
 
-		//inserts *child* lists of a parent artifact ID (not associated artifacts)
-		protected void UpdateChildListObjects<T>(IList<T> objectsToUpdate, int parentArtifactId)
-			where T : BaseDto
-		{
-			var objectsToBeInsertedLookup = objectsToUpdate.ToLookup(x => x.ArtifactId == 0);
-			var objectsToBeInserted = objectsToBeInsertedLookup[true];
-			var objectsToBeUpdated = objectsToBeInsertedLookup[false];
+				var childType = propertyInfo.PropertyType;
 
-			//insert ones that do not exist
-			if (objectsToBeInserted.Any())
-			{
-				this.InvokeGenericMethod(typeof(T), nameof(InsertChildListObjects), objectsToBeInserted, parentArtifactId);
+				this.InvokeGenericMethod(childType, nameof(InsertOrUpdate), MakeGenericList(childObjectsToUpdate, childType), recursive);
 			}
+		}
 
-			if (!objectsToBeUpdated.Any())
-				return;
-
-			var childObjectsInfo = BaseDto.GetRelativityObjectChildrenListProperties<T>();
+		private void UpdateMultipleObjectFields<T>(IList<T> objectsToUpdate, bool recursive) where T : BaseDto
+		{
+			var multipleObjectProperties =
+				typeof(T).GetProperties()
+				.Where(x => x.GetCustomAttribute<RelativityObjectFieldAttribute>()?.FieldType == RdoFieldType.MultipleObject);
 
-			//if do not have child objects in turn, we can take a shortcut
-			//and batch update all the items at once
-			if (childObjectsInfo.Count == 0)
+			foreach (var propertyInfo in multipleObjectProperties)
 			{
-				//update RDOs in bulk
-				UpdateRdos(objectsToBeUpdated.Select(x => x.ToRdo()).ToArray());
-
-				//Cannot update files in bulk; do here
-				if (typeof(T).GetProperties().ToList()
-					.Any(c => c.DeclaringType.IsAssignableFrom(typeof(RelativityFile))))
-				{
-					foreach (var objectToBeUpdated in objectsToBeUpdated)
-					{
-						InsertUpdateFileFields(objectToBeUpdated, objectToBeUpdated.ArtifactId);
-					}
-				}
+				var childObjectsToUpdate = objectsToUpdate
```

Comment:
Would it be better if we change the naming of local fields in **UpdateSingleObjectFields** & **UpdateMultipleObjectFields** to explicitly state Single or Multi object children? E.g., instead of **childObjectsToUpdate**, something like **singleObjectsToUpdate** or anything similar? As Relativity state them as Single and Multi Object fields of an object and they are not referred as actual children.

Just being curious for your opinion on this.